core/dbt/contracts/graph/unparsed.py

```diff
@@ -476,7 +476,7 @@ class UnparsedMetric(dbtClassMixin, Replaceable):
     calculation_method: str
     timestamp: str
     description: str = ""
-    expression: Union[str, int] = ""
+    expression: Union[str, int] = None
```

Comment:
@callum-mcdata I think you need to make the same change to `ParsedMetric` as well. Hoping that resolves the mypy error (though never guaranteed)

Reply:
@jtcohen6 am I misunderstanding the `ParsedMetric`? The `expression` attribute is set to `expression: str`, which makes sense because I would think that enforces that it **must** be a string. I was hoping that the change to unparsed metric would then cause a None from `UnparsedMetric` to fail the requirement of string for `ParsedMetric`

Reply:
@callum-mcdata Ah, I understand now.

Two options here:
1. Leaving it as is, but adding a check in `UnparsedMetric.validate()` that raises a nice exception if `expression` is missing/`None` (after `rename_metric_attr` is handled of course)
2. Change the type to match exactly what's in `ParsedMetric`, that is a `str` field with no default (= required):
```
    expression: str
    description: str = ""
```
Those two have to be reordered, because required fields need to come before optional fields with defaults

I don't see any `mypy` errors because the types now match exactly. And when I define a metric that's missing `expression` + `sql` (old name), This is the exception I now see:
```
dbt.exceptions.ParsingException: Parsing Error
  Invalid metrics config given in FilePath(searched_path='models', relative_path='my_metric.yml', modification_time=1663957848.4168892, project_root='/Users/jerco/dev/scratch/testy') @ metrics: {'name': 'new_customers', 'label': 'New Customers', 'model': "ref('my_model')", 'description': 'The number of paid customers using the product', 'timestamp': 'signup_date', 'time_grains': ['day', 'week', 'month'], 'dimensions': ['plan', 'country'], 'filters': [{'field': 'is_paying', 'operator': 'is', 'value': 'true'}, {'field': 'lifetime_value', 'operator': '>=', 'value': '100'}, {'field': 'company_name', 'operator': '!=', 'value': "'Acme, Inc'"}, {'field': 'signup_date', 'operator': '>=', 'value': "'2020-01-01'"}], 'calculation_method': 'count'} - at path []: 'expression' is a required property
```
Could be nicer, but unfortunately that's our standard hologram validation error :/

Do you know why we added `int` as an option in https://github.com/dbt-labs/dbt-core/pull/5027? Is expression ever an `int`?

Reply:
Got it. For sake of clean error messages, I'm leaning towards option 1 even though I don't like the idea of adding more things to the custom `validate`. Let me take a stab at this today and let you know how it goes!

As for the `int` function - I think it was to support things like "field + 1"? I'm not entirely sure though